src/plookup/table/lookup_table.rs

```diff
@@ -0,0 +1,696 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Structs and functions for LookupTables
+//! Denoted as 't' in Plookup paper.
+
+use super::hash_tables::constants::{
+    BLS_SCALAR_REAL, DECOMPOSITION_S_I, SBOX_U256,
+};
+use crate::error::Error;
+use crate::plookup::MultiSet;
+use crate::prelude::BlsScalar;
+use alloc::vec::Vec;
+
+/// For the implemenation of look up tables in PLONK, aptly named PLOOKup
+/// tables, there will be different fucntions depending on the type of table
+/// that needs to be constructed. All tables entries envisioned will be with
+/// different arity. Meaning each of the wires will correspond to a column.
+///
+/// If the standard composer calls a plookup gate, then the user will define
+/// the length of the gate, measured in circuit size.
+
+/// This struct is a table, contaning a vector,
+/// of arity 3 where each of the values is a
+/// BlsScalar. The elements of the table are
+/// determined by the function g for
+/// g(x,y), used to compute tuples.
+///
+/// This struct will be used to determine
+/// the outputs of gates within arithmetic
+/// circuits.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct PlookupTable3Arity(Vec<[BlsScalar; 3]>);
```

Comment:
The permutation implementation seems to support only arity-4. Hence, this structure can't ever be used and should be removed